src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateElectionViewModel.cs

```diff
@@ -0,0 +1,238 @@
+﻿using CommunityToolkit.Mvvm.DependencyInjection;
+using CommunityToolkit.Mvvm.Input;
+
+namespace ElectionGuard.UI.ViewModels;
+
+public partial class CreateElectionViewModel : BaseViewModel
+{
+    private readonly KeyCeremonyService _keyCeremonyService;
+    private readonly ElectionService _electionService;
+    private readonly ManifestService _manifestService;
+    private readonly ContextService _contextService;
+    private readonly ConstantsService _constantsService;
+    private const string PageName = "CreateElection";
+
+    public CreateElectionViewModel(IServiceProvider serviceProvider, KeyCeremonyService keyCeremonyService, ElectionService electionService, ManifestService manifestService, ContextService contextService, ConstantsService constantsService) : base(PageName, serviceProvider)
+    {
+        _keyCeremonyService = keyCeremonyService;
+        _electionService = electionService;
+        _manifestService = manifestService;
+        _contextService = contextService;
+        _constantsService = constantsService;
+    }
+
+    public override async Task OnAppearing()
+    {
+        KeyCeremonies = await _keyCeremonyService.GetAllCompleteAsync();
+    }
+
+    [ObservableProperty]
+    private string? _errorMessage;
+
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(CreateElectionCommand))]
+    private string? _manifestErrorMessage;
+
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(CreateElectionCommand))]
+    private string _electionName = string.Empty;
+
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(CreateElectionCommand))]
+    private KeyCeremony? _keyCeremony = null;
+
+    [ObservableProperty]
+    private List<KeyCeremony> _keyCeremonies;
+
+    [ObservableProperty]
+    private string _electionUrl;
+
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(CreateElectionCommand))]
+    private string _manifestNames;
+
+    private List<FileResult> _manifestFiles = new();
+
+    [ObservableProperty]
+    private bool _isNameEnabled = true;
+
+    [RelayCommand(CanExecute = nameof(CanCreate))]
+    private async Task CreateElection()
+    {
+        var multiple = _manifestFiles.Count > 1;
+        var lastElectionId = string.Empty;
+
+        for (var i = 0; i < _manifestFiles.Count; i++)

```

Comment:
Since these are independent operations, would this be an opportunity for us to use `Parallel.ForEach()`?

Reply:
agreed